Google.PowerShell/Container/GkeClusterCmdlets.cs

```diff
@@ -909,6 +1043,56 @@ private Cluster BuildCluster()
         }
 
         /// <summary>
+        /// Set AddonsConfig of cluster if user wants to disable horizontal pod autoscaling or HTTP load balancing.
+        /// </summary>
+        private void SetAddonsConfig(Cluster cluster, bool disablePodAutoscaling, bool disableLoadBalancing)
+        {
+            if (disablePodAutoscaling || disableLoadBalancing)
+            {
+                var addons = new AddonsConfig();
+                if (disableLoadBalancing)
+                {
+                    addons.HttpLoadBalancing = new HttpLoadBalancing() { Disabled = true };
+                }
+
+                if (disablePodAutoscaling)
+                {
+                    addons.HorizontalPodAutoscaling = new HorizontalPodAutoscaling { Disabled = true };
+                }
+
+                cluster.AddonsConfig = addons;
+            }
+        }
+
+        /// <summary>
+        /// Create a node pool based on either the NodeConfig object passed in or ByNodeConfigValues parameters.
+        /// </summary>
+        private IEnumerable<NodePool> BuildNodePools(NodeConfig nodeConfig)
+        {
+            string[] nodePoolNames = null;
+
+            if (NumberOfNodePools == null || NumberOfNodePools <= 1)
+            {
+                // By default, if there is 1 NodePool, GKE creates a NodePool with the name "default-pool".
```

Comment:
This comment confused me a little.  Is GKE creating these name or are we passing them to GKE?  Is the name optional then? and GKE will fill it in if we don't?